maven-model-builder/src/test/java/org/apache/maven/model/building/BuildModelSourceTransformerTest.java

```diff
@@ -136,6 +137,10 @@ public static boolean equalsDeep(Object m1, Object m2) {
             }
             BeanInfo bean = Introspector.getBeanInfo(m1.getClass());
             for (PropertyDescriptor prop : bean.getPropertyDescriptors()) {
+                if (("first".equals(prop.getName()) || "last".equals(prop.getName()))
```

Comment:
How is this related to building on JDK 21?

Reply:
New sequence api i guess

Reply:
Exactly, see https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/SequencedCollection.html

Those getters are throwing `NoSuchElementException` when the list is empty.